services/worker/src/worker/job_runners/config/info.py

```diff
@@ -92,11 +91,9 @@ def get_job_runner_version() -> int:
         return PROCESSING_STEP_CONFIG_INFO_VERSION
 
     def compute(self) -> CompleteJobResult:
-        if self.dataset is None:
-            raise ParameterMissingError("'dataset' parameter is required")
-        if self.config is None:
-            raise ParameterMissingError("'config' parameter is required")
-        return CompleteJobResult(compute_config_info_response(dataset=self.dataset, config=self.config))
+        return CompleteJobResult(
+            compute_config_info_response(dataset=self.dataset, config=self.config)  # type: ignore
```

Comment:
we have to find a better solution than adding `# type: ignore`, otherwise it's better to keep the code untouched.

Not sure how. Should we have DatasetJobRunner, ConfigJobRunner and SplitJobRunner, inherited from JobRunner?